server/src/main/java/com/cloud/server/ManagementServerImpl.java

```diff
@@ -2424,7 +2425,9 @@ private void setParameters(SearchCriteria<IPAddressVO> sc, final ListPublicIpAdd
             sc.setParameters("state", state);
         }
 
-        sc.setParameters( "forsystemvms", false);
+        if (sourceNetworkId == null && IpAddressManagerImpl.SystemVmPublicIpReservationModeStrictness.value()) {
+            sc.setParameters("forsystemvms", false);
+        }
```

Comment:
@SadiJr 
I have two questions
(1) if the global setting is `true`,  can the IP (in the reserved ip range  but used by uservm) listed by api ?
(2) if the global setting is `false`, can domain admin and normal users see the public ips used by systemvms ?

Reply:
@weizhouapache Sorry for the delay in responding:

1 - If the configuration is 'true', the user will not be able to use the reserved IP range. This scenario will only be true if the setting is 'false', the user creates a VM that will use one of the reserved IPs, and then the setting is again set to 'true'. In this scenario, the reserved IPs for the system VMs will not be listed by the API.

2 - No, you will only see the IPs from that range that are available, and those that are used by the VMs themselves, but the IPs used by System VMs are not listed.

Reply:
> 1 - If the configuration is 'true', the user will not be able to use the reserved IP range. This scenario will only be true if the setting is 'false', the user creates a VM that will use one of the reserved IPs, and then the setting is again set to 'true'. In this scenario, the reserved IPs for the system VMs will not be listed by the API.
> 
> 2 - No, you will only see the IPs from that range that are available, and those that are used by the VMs themselves, but the IPs used by System VMs are not listed.

@SadiJr 
thanks for reply.
2 looks good to me.
however, for 1, as you said , "In this scenario, the reserved IPs for the system VMs will not be listed by the API." so this issue is not really fixed, right ?
would it be better to use
`IpAddressManagerImpl.SystemVmPublicIpReservationModeStrictness.value() || isAllocated` ?

Reply:
@SadiJr 
no problem. thanks for your reply.
The 4.17.0.0 code freeze date is very close. If you want this PR to be included, please update.

Reply:
Done, thanks for the suggestion and sorry for the delay in answer.

Reply:
@SadiJr 
thanks for update. I will review this PR